src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/V1Deserializer.cs

```diff
@@ -77,7 +78,7 @@ private List<MetricDefinitionV1> DeserializeMetrics(YamlMappingNode rootNode, IE
         {
             if (rootNode.Children.TryGetValue("metrics", out var metricsNode))
             {
-                return _metricsDeserializer.Deserialize((YamlSequenceNode)metricsNode, errorReporter);
+                return _metricsDeserializer.Deserialize((YamlSequenceNode)metricsNode, errorReporter).ToList();
```

Comment:
```suggestion
                return _metricsDeserializer.Deserialize((YamlSequenceNode)metricsNode, errorReporter)?.ToList();
```

Reply:
Same as the other comment - I've just gone ahead and changed the type from `List<>` to `IReadOnlyCollection<>`, which means we can just get rid of the `ToList()` method call and not worry about this.